### PR TITLE
dialects: dialects: add snitch → riscv lowering

### PR DESCRIPTION
**Pull request by nazavode**
This PR adds lowering of `snitch` operations to `riscv` operations that represent the correct instruction sequences needed to actually configure streaming on the actual hw.

### CODE REVIEW

**Comment by nazavode** [comment]
Note: this is larger than the suggested patch size, a reasonable split would be one PR for the `riscv.scfgw` op and another for the actual lowering.

File: xdsl/dialects/snitch.py [hunk 1]
```diff
@@ -20,6 +22,18 @@
...
+
+
+@dataclass(frozen=True)
+class SnitchResources:
+    """
+    Bounds for resources provided by the Snitch architecture.
+    """
+
+    # Number of dimensions supported by each data mover.
+    Dimensions: int = 4
```
Comment (superlopuh) [inline xdsl/dialects/snitch.py:35]: Is this a class or instance variable?

File: xdsl/transforms/lower_snitch.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+)
+
+from xdsl.dialects.builtin import IntegerAttr, i32
+from xdsl.dialects import snitch, riscv, builtin
+
+
+@dataclass(frozen=True)
+class SnitchStreamerDimension:
+    # Offset of the streamer bound configuration for a specific dimension
+    Bound: int
```
Comment (superlopuh) [inline xdsl/transforms/lower_snitch.py:27]:
```suggestion
    bound: int
```
and below

**Review by superlopuh — CHANGES_REQUESTED** [review]

**Review by superlopuh — APPROVED** [review]

**Review by tobiasgrosser — APPROVED** [review]
This LGTM

**Pull request merged**